doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -392,8 +465,64 @@ stands for :n:`let @ident := fun {+ @binder} => @term in @term’`.
 
 .. index:: match ... with ...
 
-Definition by case analysis
----------------------------
+Definition by cases: match
+--------------------------
+
+.. insertgram term_match pattern_list_or2
+
+.. productionlist:: coq
+   term_match : match `case_item_list_comma` `return_type_opt` with `or_opt` `eqn_list_or_opt` end
+   case_item_list_comma : `case_item_list_comma` , `case_item`
+                        : `case_item`
+   return_type_opt : `return_type`
+                   : `empty`
+   as_return_type : `as_name_opt` `return_type`
+                  : `empty`
+   return_type : return `term100`
+   or_opt : |
+          : `empty`
+   case_item : `term100` `as_name_opt` `in_opt`
+   as_name_opt : as `name`
+               : `empty`
+   in_opt : in `pattern`
+          : `empty`
+   eqn_list_or_opt : `eqn_list_or`
+                   : `empty`
+   eqn_list_or : `eqn_list_or` | `eqn`
+               : `eqn`
+   eqn : `pattern_list_or` => `term`
+   pattern_list_or : `pattern_list_or` | `pattern_list_comma`
+                   : `pattern_list_comma`
+   pattern_list_comma : `pattern_list_comma` , `pattern`
+                      : `pattern`
+   pattern : `pattern10` : `term`
+           : `pattern10`
+   ident_opt : `ident`
+             : `empty`
+   pattern10 : `pattern1` as `name`
+             : `pattern1` `pattern1_list`
+             : @ `qualid` `pattern1_list_opt`
+             : `pattern1`
+   pattern1_list : `pattern1_list` `pattern1`
+                 : `pattern1`
+   pattern1_list_opt : `pattern1_list_opt` `pattern1`
+                     : `empty`
+   pattern1 : `pattern0` % `ident`
+            : `pattern0`
+   pattern0 : `qualid`
+            : {| `record_patterns` |}
+            : _
+            : ( `pattern` )
+            : ( `pattern` | `pattern_list_or2` )
+            : `numeral`
+            : `string`
+   record_patterns : `record_pattern` ; `record_patterns`
+                   : `record_pattern` ;
+                   : `record_pattern`
+                   : `empty`
+   record_pattern : `qualid` := `pattern`
+   pattern_list_or2 : `pattern_list_or2` | `pattern`
```

Comment:
This one should be the one to be called `pattern_list_or`. The previous `pattern_list_or` should rather be called `pattern_list_comma_list_or`.